### PR TITLE
Add /healthcheck.json endpoint

### PR DESCRIPTION
**Pull request by threedaymonk**
This exposes a JSON document at `/healthcheck.json` that reports the state of queues, database connectivity, SMTP server, and overall health of the application.

### CODE REVIEW

**Comment by SteveMarshall** [comment]
Might be nice to add a check for connecting to SendGrid and reporting that in the health check, too, to give us a more direct indicator of if the SMTP connection is down.

**Comment by SteveMarshall** [comment]
I wonder if it's worth adding more explicit error messages for some of the compound checks here? For example, making it explicit that we couldn't connect to the queue in `queue_info`.

File: app/services/healthcheck.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  def queue_info(name, description)
+    q = Sidekiq::Queue.new(name)
+    {
+      description: description,
+      ok: fresh?(q),
+      oldest: oldest(q),
+      count: q.count
+    }
+  rescue StandardError
+    { description: description, ok: false, oldest: nil, count: 0 }
```
Comment (SteveMarshall) [inline app/services/healthcheck.rb]: Can we offer a more explicit “do not trust these values” value for  `count`? (Perhaps `nil` as in `oldest`, or maybe that should have a more explicit one, too…)

**Comment by threedaymonk** [comment]
@SteveMarshall It now reports any errors it catches, and I've added an SMTP connection check.

File: app/services/healthcheck/check_component.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  private
+
+    def build_report(description)
+      @report =
+        begin
+          yield
+        rescue StandardError => err
+          { error: err.to_s, ok: false }
+        end.merge(description: description)
```
Comment (SteveMarshall) [inline app/services/healthcheck/check_component.rb]: This is really neat! :+1:

**Comment by SteveMarshall** [comment]
LGTM; @tatyree?

File: app/services/healthcheck/check_component.rb (hunk 2 above)
Comment (tatyree) [inline app/services/healthcheck/check_component.rb]: I agree .

**Pull request merged**